tests/boot/boot_from_pxe.pm

```diff
@@ -45,6 +45,9 @@ sub run {
             select_console 'sol', await_console => 0;
         }
     }
+    if (!check_screen([qw(virttest-pxe-menu qa-net-selection prague-pxe-menu pxe-menu)], 600)) {
```

Comment:
Will it be more robust if making it in a while loop to try several times?

Reply:
serial console should finish initialization in about 5 minutes. And then pxe boot menu may passed away. If so, it will trigger `ipmi power reset` in about 5 min(because check_screen timeout is 600). But serial won't reinitialize due to we keep console connection. That's to say serial must works after first power reset, or else never works even with while loop.serial console should finish initialization in about 5 minutes. And then pxe boot menu may passed away. If so, it will trigger `ipmi power reset` in about 5 min(because check_screen timeout is 600). But serial won't reinitialize due to we keep console connection. That's to say serial must works after first power reset, or else never works even with while loop.

Reply:
This code enhancement is based on the assumption that serial console sol works after "power reset", right? Then making them in a loop should improve more. But given that one loop takes a booting cycle (several minutes), trying once only can be a good balance point.